clang/lib/CodeGen/CGLoopInfo.h

```diff
@@ -111,11 +111,8 @@ struct LoopAttributes {
   /// Value for llvm.loop.ii.count metadata.
   unsigned SYCLIInterval;
 
-  /// Flag for llvm.loop.max_concurrency.count metadata.
-  bool SYCLMaxConcurrencyEnable;
-
-  /// Value for llvm.loop.max_concurrency.count metadata.
-  unsigned SYCLMaxConcurrencyNThreads;
+  /// Value for max_concurrency variant and metadata.
+  llvm::SmallVector<unsigned int, 2> SYCLMaxConcurrencyNThreads;
```

Comment:
Sorry for not noticing this before, but I think all of these vectors should be using `1` instead of `2` -- there's very little chance someone's going to add two attributes here, right? Actually, why does this need to be a vector at all? Should this be using an `llvm::Optional<unsigned>` instead?

Reply:
>>Should this be using an llvm::Optional<unsigned> instead?

Yes, we can use this instead. Thanks for the suggestion.